structure/heap/heap.go

```diff
@@ -0,0 +1,80 @@
+// Package heap : max heap implemnation
+// https://en.wikipedia.org/wiki/Heap_(data_structure)
+package heap
```

Comment:
Could you look into our [this section](https://github.com/TheAlgorithms/Go/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) in our contribution guideline.

```suggestion
// filename
// description: Add one line description here
// details:
// This is a multi-line
// description containing links, references,
// math equations, etc.
// author(s) [Name](https://github.com/handle), [Name](https://github.com/handle)
// see relatedfile.go, anotherfile.go, file_test.go

// Package heap implements a max heap
// https://en.wikipedia.org/wiki/Heap_(data_structure)
package heap
```